Assets/Scripts/Player/PlayerInventory.cs

```diff
@@ -189,6 +198,7 @@ void SpawnBuddy()
 
 		inventory[heldResources[resourceIndex]]--;
 		UpdateResourceList();
+		return true;
```

Comment:
I feel like this function would make more sense logically if this ordering was reversed. if(WithinCylinder) {SpawnBuddy; return true; } else { return false; }